pkg/chunk/cassandra/storage_client.go

```diff
@@ -121,6 +121,27 @@ func (cfg *Config) createKeyspace() error {
 	return errors.WithStack(err)
 }
 
+// PasswordAuthenticator creates the gocql Authenticator object
+type PasswordAuthenticator struct {
+	Username string
+	Password string
+}
+
+// Challenge implements the Authenticator interface for gocql
+func (p PasswordAuthenticator) Challenge(req []byte) ([]byte, gocql.Authenticator, error) {
+	resp := make([]byte, 2+len(p.Username)+len(p.Password))
+	resp[0] = 0
```

Comment:
Is there some reference for the particular format of this byte slice? Might be good to put that in a comment.